lib/openssl/ssl.rb

```diff
@@ -276,7 +277,10 @@ def verify_certificate_identity(cert, hostname)
             if san.value.size == 4
               return true if san.value.unpack('C*').join('.') == hostname
             elsif san.value.size == 16
-              return true if san.value.unpack('n*').map { |e| sprintf("%X", e) }.join(':') == hostname
+              begin
+                return true if IPAddr.new(san.value.unpack('n*').map {|e| sprintf("%X", e)}.join(':')) == IPAddr.new(hostname)
+              rescue IPAddr::InvalidAddressError
+              end
             end
```

Comment:
I guess this can be simplified to `return true if san.value == IPAddr.new(hostname).hton`.

Reply:
Much nicer, and shorter. I have updated my PR.

Note: I could not remove the  `begin ... rescue` guard: I tried, and I got the  the following failure:
```
Error: test_verify_certificate_identity(OpenSSL::TestSSL): IPAddr::InvalidAddressError: invalid address
...
408 tests, 3850 assertions, 0 failures, 1 errors, 5 pendings, 0 omissions, 0 notifications
98.5294% passed
```